fs/move.py

```diff
@@ -55,6 +59,31 @@ def move_file(
             resources (defaults to `False`).
 
     """
+    if src_fs.hassyspath(src_path) and dst_fs.hassyspath(dst_path):
+        # we have to raise a ResourceReadOnly exception manually if a FS is read-only
+        if src_fs.getmeta().get("read_only", True):
+            raise ResourceReadOnly(src_fs, src_path)
+        if dst_fs.getmeta().get("read_only", True):
+            raise ResourceReadOnly(dst_fs, dst_path)
+
+        # if both filesystems have a syspath we create a new OSFS from a
+        # common parent folder and use it to move the file.
+        with manage_fs(src_fs) as _src_fs:
+            with manage_fs(dst_fs, create=True) as _dst_fs:
```

Comment:
Should these `manage_fs` calls (and also the ones elsewhere in this function) all be setting `writeable=True` ? (since if we're moving a file, both the `src_fs` and `dst_fs` will end up getting modified).

Reply:
@althonos On a related note, why does the `move_dir` function elsewhere in this file use `manage_fs(src_fs, writeable=False)` ? Does that mean that theoretically the `_src_fs.removetree(src_path)` at the end of that function shouldn't work?

Reply:
Yes that's a bug. It's possible to have a file copied over but the src is not removed. I just submitted a test case for that and try to fix this.

Reply:
Just settings these to `writeable=True` in `manage_fs` isn't enough. I'm unsure, is this a bug? Should I just delete the copy when deleting the source fails?

Reply:
I submitted a proposal which cleans up correctly after itself.

Reply:
@althonos On _another_ sidenote, perhaps https://github.com/PyFilesystem/pyfilesystem2/blob/f6a6195d10452da79b0068a7003042fa03c87ab4/fs/opener/registry.py#L285-L286 should check that the FS-object is writable if `writeable=True` is passed in, which I think would also have the side-benefit of tidying up the code in this PR?

Reply:
Yes, I kind of expected it to do this. 👍

Reply:
I implemented this and cleaned up the PR.